libtsuba/include/tsuba/FileFrame.h

```diff
@@ -76,6 +76,11 @@ class KATANA_EXPORT FileFrame : public arrow::io::OutputStream {
     return reinterpret_cast<T*>(map_start_); /* NOLINT */
   }
 
+  /// only data up to cursor is written out, flavors of Write
+  /// automatically track this, so SetCursor is only useful when
+  /// treating the FileFrame like a buffer (e.g., using ptr())
+  katana::Result<void> SetCursor(uint64_t new_cursor);
```

Comment:
I'm not sure why I always find FileFrames confusing but is this new interface only for testing?  If so, it should be commented as a test only interface.

Reply:
It's not only for testing. It's an interface that lets you use `FIleFrame` as a buffer. Until this patch it's only been used as a stream.

Not saying it should be this way but currently: `FileFrame` has some mapped memory, and copies things into it as you call `Wrtie`.  When you call write it bumps the cursor and when things are persisted they are only persisted up to the cursor.